neo/Consensus/ConsensusContext.cs

```diff
@@ -95,6 +94,16 @@ public ConsensusPayload MakeChangeView()
             });
         }
 
+        public ConsensusPayload MakeCommit()
+        {
+            if (Commits[MyIndex] == null)
+                Commits[MyIndex] = MakeHeader()?.Sign(keyPair);
```

Comment:
If `MakeHeader` return `null` the serialization of a null Signature inside the `Commit` message could produce a errors.

Reply:
It is impossible. Because we have already received the `PrepareRequest`, so we have enough information to build the block header.

Reply:
The only place to call `MakeCommit()` is here:

https://github.com/neo-project/neo/blob/9640c497f09fb1e72c8e61b794f4c0da5b8bfef2/neo/Consensus/ConsensusService.cs#L106-L116

The `if` statement here can guarantee that we call `MakeHeader()` without returning `null`.